lib/common_fw/common_fw.cpp

```diff
@@ -9,18 +9,20 @@ SdFat sd;
 File file;
 struct Queue *sdBuffer;
 union floatArrToBytes farrbconvert;
+char packet[64];
+char command[64];
 
 /**
  *
  */
-bool write_to_SD(std::string message, const char * file_name) {
-    sdBuffer->enqueue(message);
+bool write_to_SD(uint8_t *message, uint8_t size, const char * file_name) {
+    sdBuffer->enqueue(message, size);
     if(sdBuffer->length >= 40) {
       if(file.open(file_name, O_RDWR | O_APPEND)) {
```

Comment:
If it's not open, shouldn't this function still pop an item from the front of the queue to prevent it from growing indefinitely?

Reply:
No I don't think so. I think we should let it fail so that we know there isn't an additional and then put an SD in.